csrc/id_model/schedule.cpp

```diff
@@ -10,10 +10,48 @@
 
 namespace nvfuser {
 
+IterDomain* representativeId(const ValGroup& vg) {
+  IterDomain* rep = nullptr;
+
+  auto preferNewIterType = [&rep](IterDomain* new_id) {
+    if (rep->isReduction()) {
+      return new_id->isBroadcast() || new_id->isIteration();
+    } else if (rep->isBroadcast()) {
+      return new_id->isIteration();
+    } else if (rep->isIteration()) {
+      return false;
+    } else {
+      // Prefer anything else to a non-iter, non-bcast, non-reduction ID
+      return true;
+    }
+  };
+
+  auto preferNewExtent = [&rep](IterDomain* new_id) {
+    if (rep->hasExpandedExtent() && !new_id->hasExpandedExtent()) {
```

Comment:
If we have:

ID1: extent 1, expanded extent 10
ID2: extent 1, no expanded extent

are we preferring ID2?

Reply:
Good point, I might have had this backwards. If there are only `Broadcast` domains in the group and one is expanded, the expanded one carries more information. At first I was viewing this as "which IterDomain is most compatible with transforms" and thought expanded extents could potentially cause problems. But I think any such problems have been fixed for a while and it's better to return those expanded IDs so that we could inspect their extents. I'll reverse this condition.